core/backend/src/BlobContainerService.ts

```diff
@@ -87,21 +87,29 @@ export namespace BlobContainer {
   /** The URI and Id of the container. */
   export interface UriAndId {
     baseUri: string;
-    id: ContainerId;
+    containerId: ContainerId;
   }
 
   /** Information required to access an existing container. */
-  export interface AccessContainerProps {
-    address: UriAndId;
+  export interface AccessContainerProps extends UriAndId {
     userToken: UserToken;
   }
 
+  /**
+   * Access level to request for token.
+   * - `"write"`: request a writeable AccessToken.
+   * - `"read"`:  request a read-only AccessToken.
+   * - `"admin"`:  request a admin AccessToken.
+   * - `"writeIfPossible"`: first request a writeable AccessToken. If the user is not authorized for write, request a read-only AccessToken.
+   * If the user is not authorized for the level requested, an exception is thrown.
+   */
+  export type RequestAccessLevel = "write" | "read" | "admin" | "writeIfPossible";
```

Comment:
What does "admin" mean in this context?

For Azure, these "tokens" are really SAS URLs, right?  What rights would an "admin" SAS have that a "write" SAS wouldn't?

Reply:
I thought that would be used the delete versions if we enable automatic versioning.